contracts/interchain-token-service/src/flow_limit.rs

```diff
@@ -42,6 +44,43 @@ impl FlowDirection {
 
         env.storage().temporary().set(&key, &new_flow);
     }
+
+    /// Adds flow amount in the specified direction (in/out) for a token.
+    /// Flow amounts are stored in temporary storage since they only need to persist for
+    /// the 6-hour epoch duration.
+    ///
+    /// Checks that:
+    /// - Flow amount doesn't exceed the flow limit
+    /// - Adding flows won't cause overflow
+    /// - Total flow in one direction doesn't exceed flow in opposite direction plus limit
+    pub fn add_flow(
+        &self,
+        env: &Env,
+        token_id: BytesN<32>,
+        flow_amount: i128,
+    ) -> Result<(), ContractError> {
+        let Some(flow_limit) = flow_limit(env, token_id.clone()) else {
+            return Ok(());
+        };
+
+        let flow_to_add = self.flow(env, token_id.clone());
+        let flow_to_compare = self.reverse_flow(env, token_id.clone());
+
+        ensure!(flow_amount <= flow_limit, ContractError::FlowLimitExceeded);
+        let new_flow = flow_to_add
+            .checked_add(flow_amount)
+            .ok_or(ContractError::FlowLimitExceeded)?;
+        let max_allowed = flow_to_compare
+            .checked_add(flow_limit)
+            .ok_or(ContractError::FlowLimitExceeded)?;
+        ensure!(new_flow <= max_allowed, ContractError::FlowLimitExceeded);
```

Comment:
```suggestion
        ensure!(flow_amount <= flow_limit, ContractError::FlowLimitExceeded);

        let new_flow = self.flow(env, token_id.clone())
            .checked_add(flow_amount)
            .ok_or(ContractError::FlowLimitExceeded)?;
        let max_allowed = self.reverse_flow(env, token_id.clone())
            .checked_add(flow_limit)
            .ok_or(ContractError::FlowLimitExceeded)?;

        // Equivalent to flow_amount + flow - reverse_flow <= flow_limit
        ensure!(new_flow <= max_allowed, ContractError::FlowLimitExceeded);
```
remove unnecessary variables and add comment to explain logic